### PR TITLE
Allow to set custom REDIS keys for security endpoint and PSK ID

### PR DESCRIPTION
**Pull request by aliakseiz**
Added setters for `SEC_EP` and `PSKID_SEC` to be able to override the default values.

### CODE REVIEW

**Comment by sbernard31** [comment]
Hi, thx for this PR. :pray: 

Before to talk about the legal agreement issue :point_up: and review the code, could you explain why you need this, maybe you can explain a little your use case and the context where you need it ? This should help to provide a solution for your need.

**Comment by aliakseiz** [comment]
Hi,
we have a strict REDIS keys naming rules in our company and the default `SEC#EP#` and `PSKID#SEC` don't comply with it. Also I couldn't find a way to override them using current code base.

For the moment I use a custom `RedisSecurityStore`, which differs only by changes provided in the PR.

Wanted to avoid such a copy-pasting and to make life a bit easier when upgrading the gateway to the next milestone. Was not sure what is the correct way to do that, decided to create a PR.

By the way, ECA is settled now.

**Comment by jvermillard** [comment]
Looks useful, I'm starting to mix Leshan data structure and some other applications in the same Redis cluster, and I have the same need: a way to change the prefix to organize the data better.

That said, I'm not sure this should still be static and set by a setter, but a final attribute you can set using a constructor.

**Comment by aliakseiz** [comment]
@jvermillard thank you for suggestion, constructor is much better, indeed.
I updated PR accordingly.

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java [hunk 1]
```diff
@@ -40,16 +40,24 @@
  */
 public class RedisSecurityStore implements EditableSecurityStore {
 
-    private static final String SEC_EP = "SEC#EP#";
+    private final String SEC_EP;
```
Comment (jvermillard) [inline leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:43]: Capitalization is only for constants

Comment (aliakseiz) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java]: Sure thing, missed that somehow.

**Comment by sbernard31** [comment]
> we have a strict REDIS keys naming rules in our company and the default SEC#EP# and PSKID#SEC don't comply with it. 

OK I understand.
This makes me think to this issue : https://github.com/eclipse/leshan/issues/1249.
Do you also need this kind of feature for `RedisRegistrationStore` ?
@aliakseiz, just by curiosity you are using Leshan **v2.0.0-Mx** ?

> That said, I'm not sure this should still be static and set by a setter, but a final attribute you can set using a constructor.

:+1: 

> By the way, ECA is settled now.

:+1:

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java [hunk 2]
```diff
@@ -40,22 +40,30 @@
...
 
     private final Pool<Jedis> pool;
 
     private final List<SecurityStoreListener> listeners = new CopyOnWriteArrayList<>();
 
     public RedisSecurityStore(Pool<Jedis> pool) {
         this.pool = pool;
+        this.secEp = "SEC#EP#";
+        this.pskIdSec = "PSKID#SEC";
+    }
```
Comment (sbernard31) [inline leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:55]: Probably better to use : 

```suggestion
    public RedisSecurityStore(Pool<Jedis> pool) {
        this(pool, "SEC#EP#", "PSKID#SEC");
    }
```

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java [hunk 3]
```diff
@@ -40,22 +40,30 @@
  */
 public class RedisSecurityStore implements EditableSecurityStore {
 
-    private static final String SEC_EP = "SEC#EP#";
+    private final String secEp;
 
-    private static final String PSKID_SEC = "PSKID#SEC";
+    private final String pskIdSec;
```
Comment (sbernard31) [inline leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:45]: I don't know if you agree with me but I feel this attribute names are not so clear.
Maybe good opportunity to change it to something clearer ?

`secEp` => `securityInfoByEndpointPrefix`
`pskIdSec` => `endpointByIdentityKey`

If we change this we need to also change it for constructor arguments at : 
`public RedisSecurityStore(Pool<Jedis> pool, String secEp, String pskIdSec)`

**Comment by sbernard31** [comment]
Maybe out of scope of this PR, but I also feel that default redis key value are not so good...

**Being more consistent :** 

**1)  `SEC#EP#` key**
This is a prefix to create a key from "endpoint name" to retrieve a `SecurityInfo` (E.g. of key `SEC#EP#my_endpoint`)
(this is used with redis [GET](https://redis.io/commands/get/) command)
So first part "SEC" is what we search and second part "EP" is the field used as index and third part "my_endpoint" is the value of the field.

**2) `PSKID#SEC`** key
Here this is the key search a value in a HashStored by field (See redis [HGET](https://redis.io/commands/hget/) command)
This allow to find "endpoint name" by "PSK Identity".
So order seems different, first part is the field index name "PSKID" and second part "SEC" the value ?
But why "SEC" as we store the "endpoint name" ? 
If we want to be consistent this should be some thing like EP#PSKID ?

What do you think ? Should we at least change `PSKID#SEC` to `EP#PSKID` ?
Or any better ideas for those key name ?
Or better to not change it for backward compatibility ? (knowing that is OK to break compatibility between 1.x and 2.x and with this new PR it is easy to use old prefix)

**Should we add more prefix to all key   ?** 

Do we need a Leshan global prefix used for all store ? e.g. : `LSHN#PSKID#SEC` and` LSHN#SEC#EP#`
OR prefix by store ? e.g. : `SECSTORE#PSKID#SEC` and `SECSTORE#SEC#EP#`
OR both ? e.g. `LSHN#SECSTORE#PSKID#SEC` and `LSHN#SECSTORE#SEC#EP#`

This allow to have separate namespaces at cost of more bytes. Don't know if we should go in that way :shrug:

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java (hunk 2 above)
Comment (aliakseiz) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java]: Sure, thank you.

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java (hunk 3 above)
Comment (aliakseiz) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java]: Totally agree, more descriptive names will be better.

**Comment by aliakseiz** [comment]
> just by curiosity you are using Leshan v2.0.0-Mx ?

Yes, we have a gateway in production environment based on a v2.0.0-M10. Trying to keep it up to date with the latest Leshan since v1.0.0-RC2. Didn't face any critical issues so far.

> This makes me think to this issue : #1249.
> Do you also need this kind of feature for `RedisRegistrationStore` ?

Not really, but I realize that for sake of consistency and convenience it might be nice to have an ability to customize REDIS keys or at least a root prefix.
I could try extending current PR with a similar approach for `RedisRegistrationStore`. Or you would suggest to prepare a separate one?

**Comment by sbernard31** [comment]
> Yes, we have a gateway in production environment based on a v2.0.0-M10. Trying to keep it up to date with the latest Leshan since v1.0.0-RC2. Didn't face any critical issues so far.

Good to know :slightly_smiling_face:  
Did you test `master` since we made lot of changes about Transport Layer Abstraction ? See https://github.com/eclipse/leshan/issues/1025#issuecomment-1373824399.

> I could try extending current PR with a similar approach for RedisRegistrationStore. Or you would suggest to prepare a separate one?

I think this is better to limit this PR to `SecurityStore`.
Then we can do a separated PR for RegistrationStore. (It will probably lead to more question because there is a lot of argument and so maybe we need a builder or another solution ?)

Maybe we should also decide about https://github.com/eclipse/leshan/pull/1398#issuecomment-1429465578 before to move forward on RegistrationStore ? :thinking:

**Comment by sbernard31** [comment]
Maybe if we need to discuss about RedisRegistrationStore we can use https://github.com/eclipse/leshan/issues/1249.

**Comment by jvermillard** [comment]
@sbernard31 if you write a migration script, I'm fine with the rename to EP#PSKID :joy:

**Comment by aliakseiz** [comment]
> Did you test master since we made lot of changes about Transport Layer Abstraction ?

Yes, no issues so far. Migration was rather easy and straightforward.

Regarding https://github.com/eclipse/leshan/pull/1398#issuecomment-1429465578:
> What do you think ? Should we at least change PSKID#SEC to EP#PSKID ?

As for me, I like the `SEC#EP#key` and `EP#PSKID` names. The order is consistent and has some logic behind. Once you get it, you can navigate easily.

> Or better to not change it for backward compatibility?

Those who will experience  issues with the new keys could easily override them to current ones and perform the migration later if needed.

> Should we add more prefix to all key ? Do we need a Leshan global prefix used for all store?

Let's indeed continue the discussion in https://github.com/eclipse/leshan/issues/1249 .

**Comment by sbernard31** [comment]
> Yes, no issues so far. Migration was rather easy and straightforward.

Thx a lot for this feedback. :pray:  
If you have any feedback or idea to improve about the new API, do not hesitate to share.  

About `EP#PSKID`, I agree with so let's change the key value.
Let's me know if you plan to do it in this PR ?
If yes, maybe in another commit, with javadoc explaining the break between v1.x and v2.x and old key to use for backward compatibility ?

> Let's indeed continue the discussion in https://github.com/eclipse/leshan/issues/1249 .

I'm waiting for your opinion about prefix. :slightly_smiling_face:

**Comment by aliakseiz** [comment]
> Let's me know if you plan to do it in this PR ?
> If yes, maybe in another commit, with javadoc explaining the break between v1.x and v2.x and old key to use for backward compatibility ?

Updated the PR accordingly. Any suggestions on the wording in javadoc are welcome.

> Do we need a Leshan global prefix used for all store ? e.g. : LSHN#PSKID#SEC and LSHN#SEC#EP#
> OR prefix by store ? e.g. : SECSTORE#PSKID#SEC and SECSTORE#SEC#EP#
> OR both ? e.g. LSHN#SECSTORE#PSKID#SEC and LSHN#SECSTORE#SEC#EP#

From my point of view, if you have a Redis instance dedicated for Leshan, prefix by store makes the most sense. Because it provides a necessary grouping and a quick overview of what is stored under those keys.

Global prefix might be useful only when Redis is shared between services, but in this case the prefix `LSHN#SECSTORE#SEC#EP#` most probably will not comply with your internal naming rules and you would want to use a custom prefixes anyway.

**Comment by sbernard31** [comment]
> Any suggestions on the wording in javadoc are welcome

I review the last commit and it looks good to me. :+1: 

Just a small detail about the changes on `CONTRIBUTING.md`, in the future do not hesitate to create dedicated commit or even dedicated PR  for this kind of "out ot topic" changes.

> From my point of view, if you have a Redis instance dedicated for Leshan, prefix by store makes the most sense. Because it provides a necessary grouping and a quick overview of what is stored under those keys.
> 
> Global prefix might be useful only when Redis is shared between services, but in this case the prefix LSHN#SECSTORE#SEC#EP# most probably will not comply with your internal naming rules and you would want to use a custom prefixes anyway.

It makes sense to me. So let's add only 1 store prefix.

Remaining questions are : 
1. What should we use as default prefix ? `SECSTORE#` or `LSHNSECSTORE` or something else ?
2. Do we want to add a way to set prefix ? (like adding new constructor or add parameter to existent one)  OR letting user able to only change key is enough.

If we want to provide a way to set prefix we need to decide how ?  none exclusive solution could be : 
2.1 add a constructor `RedisSecurityStore(Pool<Jedis> pool, String prefix)`
2.2 add a constructor `RedisSecurityStore(Pool<Jedis> pool, String prefix, String securityInfoByEndpointPrefix, String endpointByPskIdKey)`
2.3 add a builder.

For prefix do you prefer to continue with this PR or a new one ?
Just by curiosity do you also plan to work on `RedisRegistrationStore` ?

Again thx for you contribution, that's really appreciated :pray:

**Comment by aliakseiz** [comment]
> in the future do not hesitate to create dedicated commit

Sorry about that, probably a bad habit of mine from other projects. Will keep it in mind next time.

I would go for `SECSTORE#`, because it's shorter and still clear enough. If somebody would need to have an indication that it belongs to Leshan, it can be easily overridden via constructor.
To me it looks like using a builder here might be a bit to excessive. Not sure if the builders are widely used across Leshan and are treated as a common practice.
Maybe options 2.1 and 2.2 would be enough? First one for those who don't care about the internal naming and just need to hide the Leshan security store under some key. And second one for complete control over the keys and prefixes.

I will update this PR with prefix setting.

>  Just by curiosity do you also plan to work on RedisRegistrationStore ?

Yes, right after finishing current PR.

**Comment by sbernard31** [comment]
> Sorry about that,

Not a big deal :wink: 

> I would go for SECSTORE#

:+1: 

> Not sure if the builders are widely used across Leshan and are treated as a common practice.

I can confirm Leshan code base has several Builder. There are generally added when there is too many parameters in constructors or too many constructors.
I know `Builder` has many drawback but I didn't find better way to create instance of a class with many `final` parameters.

> To me it looks like using a builder here might be a bit to excessive.

I agree adding  builder for `RedisSecurityStore` seems a bit overkill.
But my guess is that for `RedisRegisrationStore` this will be hard to avoid `Builder` because there is so many parameters.
So one reason to use Builder in `RedisSecutiryStore` could be for API consistency with `RedisRegisrationStore` (of course this point makes sense only if we decide to use builder in `RedisRegisrationStore`)

I just said that to let you know the big picture.  
On my side, I have no strong opinion between `builder` vs `more constructor`  in `RedisSecurityStore`.
So, feel free to add `constructor` or builder. 

> Maybe options 2.1 and 2.2 would be enough? First one for those who don't care about the internal naming and just need to hide the Leshan security store under some key. And second one for complete control over the keys and prefixes.

If you chose constructor, this makes sense to me.

> Yes, right after finishing current PR.

Great :+1:

**Comment by aliakseiz** [comment]
> ...for API consistency with RedisRegisrationStore

That was exactly what I thought about right after sending my previous comment :slightly_smiling_face: 
Had some doubts regarding usage mostly because I am not familiar with the Leshan codebase yet. 

> because there is so many parameters

I totally agree with you and will add a builder then.

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java [hunk 4]
```diff
@@ -52,14 +49,10 @@ public class RedisSecurityStore implements EditableSecurityStore {
...
 
-    public RedisSecurityStore(Pool<Jedis> pool) {
-        this(pool, "SEC#EP#", "EP#PSKID");
-    }
-
-    public RedisSecurityStore(Pool<Jedis> pool, String securityInfoByEndpointPrefix, String endpointByPskIdKey) {
-        this.pool = pool;
-        this.securityInfoByEndpointPrefix = securityInfoByEndpointPrefix;
-        this.endpointByPskIdKey = endpointByPskIdKey;
+    private RedisSecurityStore(Builder builder) {
```
Comment (sbernard31) [inline leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:52]: If we want to allow inheritance of `RedisSecurityStore` maybe we should at least make it protected ?

**Comment by aliakseiz** [comment]
Thank you @sbernard31 , those are all valid points and suggestions.
I think I fixed all of them, please take a look again.

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java [hunk 5]
```diff
@@ -49,7 +49,13 @@ public class RedisSecurityStore implements EditableSecurityStore {
 
     private final List<SecurityStoreListener> listeners = new CopyOnWriteArrayList<>();
 
-    private RedisSecurityStore(Builder builder) {
+    public RedisSecurityStore(Pool<Jedis> pool) {
+        this.pool = pool;
+        this.securityInfoByEndpointPrefix = "SECSTORE#SEC#EP#";
+        this.endpointByPskIdKey = "SECSTORE#EP#PSKID";
```
Comment (sbernard31) [inline leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:55]: maybe you can use 
```java
this(new Builder(pool).build());
```

So there is only 1 place where we define default value.
I know we will create a builder instance  just to do that but this is something which is not called often

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java [hunk 6]
```diff
@@ -215,7 +221,7 @@ public void setEndpointByPskIdKey(String endpointByPskIdKey) {
          * Set the prefix for all keys and prefixes including {@link #securityInfoByEndpointPrefix} and
          * {@link #endpointByPskIdKey}.
          * <p>
-         * Default value is {@literal SECSTORE#}.
+         * Default value is {@literal SECSTORE#}. Should not be {@code null}, can be empty.
```
Comment (sbernard31) [inline leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:224]: Here it is said : `Should not be {@code null}`

But looking at : 
```java
    if (this.prefix != null) {
                this.securityInfoByEndpointPrefix = this.prefix + this.securityInfoByEndpointPrefix;
                this.endpointByPskIdKey = this.prefix + this.endpointByPskIdKey;
            }
```
You manage `null`.

I think it's OK to support `null` prefix.

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/util/RedisSecureIntegrationTestHelper.java [hunk 7]
```diff
@@ -34,7 +34,7 @@ protected SecurityStore createSecurityStore() {
         } else {
             jedis = new JedisPool();
         }
-        securityStore = new RedisSecurityStore(jedis);
+        securityStore = new RedisSecurityStore.Builder(jedis).build();
```
Comment (sbernard31) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/util/RedisSecureIntegrationTestHelper.java:37]: maybe we can now avoid to do this changes as constructor is back.

File: leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java [hunk 8]
```diff
@@ -162,7 +162,7 @@ public static LeshanServer createLeshanServer(LeshanServerDemoCLI cli) throws Ex
             securityStore = new FileSecurityStore();
         } else {
             // use Redis Store
-            securityStore = new RedisSecurityStore(cli.main.redis);
+            securityStore = new RedisSecurityStore.Builder(cli.main.redis).build();
```
Comment (sbernard31) [inline leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java:165]: Same here : maybe we can now avoid to do this changes as constructor is back.

File: leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java [hunk 9]
```diff
@@ -190,8 +190,10 @@ public CaliforniumServerEndpointFactory createDefaultEndpointFactory(URI uri) {
                 return new CoapsServerEndpointFactory(uri) {
 
                     @Override
-                    protected Builder createDtlsConnectorConfigBuilder(Configuration endpointConfiguration) {
-                        Builder dtlsConfigBuilder = super.createDtlsConnectorConfigBuilder(endpointConfiguration);
+                    protected DtlsConnectorConfig.Builder createDtlsConnectorConfigBuilder(
+                            Configuration endpointConfiguration) {
+                        DtlsConnectorConfig.Builder dtlsConfigBuilder = super.createDtlsConnectorConfigBuilder(
+                                endpointConfiguration);
```
Comment (sbernard31) [inline leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java:196]: I guess this part of code should not be changed ?

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java (hunk 5 above)
Comment (aliakseiz) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:55]: Certainly makes sense. I added a `protected` constructor to be able to use `this(...)` here.

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java (hunk 6 above)
Comment (aliakseiz) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:224]: Forgot about it, wrote a JavaDoc first, then added a null-check.

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/util/RedisSecureIntegrationTestHelper.java (hunk 7 above)
Comment (aliakseiz) [reply leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/util/RedisSecureIntegrationTestHelper.java:37]: Yes, of course, you're right.

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java (hunk 6 above)
Comment (sbernard31) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:224]: No problem, this happens. An easy way to limit this kind of issue is to do a review of you code, like if you review someone else's code. :wink:

File: leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java (hunk 9 above)
Comment (aliakseiz) [reply leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java:196]: Perhaps it was corrected by my IDE. Reverted it back.

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java (hunk 6 above)
Comment (aliakseiz) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:224]: This is actually one of the "rules" in my team at work. Don't know, why I didn't follow it while preparing this PR.
Will pay more attention next time.

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java (hunk 5 above)
Comment (sbernard31) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java]: Uups my mistake I wanted to say : 
``` java
this(new Builder(pool)); 
```

So no need new constructor;

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java [hunk 10]
```diff
@@ -167,4 +180,90 @@ public void addListener(SecurityStoreListener listener) {
...
+        private String endpointByPskIdKey;
+
+        private String prefix;
+
+        /**
+         * Set the Redis connection pool for the {@link RedisSecurityStore}.
+         */
+        public void setPool(Pool<Jedis> pool) {
+            this.pool = pool;
+        }
```
Comment (sbernard31) [inline leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:204]: As we give pool as Builder constructor maybe we don't need this setter or do you have a use case in mind ?

**Comment by sbernard31** [comment]
I didn't notice that you Builder is not fluent.
Are you OK if we change it in that way ? (each setter must return the builder so you can write)

```java
securityStore = new RedisSecurityStore.Builder(jedis).
       .setPrefix(null) //
       .setEndpointByPskIdKey("EPKEY") //
       .setSecurityInfoByEndpointPrefix("SECKey").build();
```

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java (hunk 5 above)
Comment (aliakseiz) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java]: Good, it's exactly what I wanted to do at first.

File: leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java (hunk 10 above)
Comment (aliakseiz) [reply leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisSecurityStore.java:204]: Good point, I removed this setter.

**Comment by aliakseiz** [comment]
> I didn't notice that you Builder is not fluent.

Don't know what I was thinking of, of course setters should return `Builder` instance. Fixed it.

**Review by sbernard31 — APPROVED** [review]
:heavy_check_mark:  I think we're good now.

I play a bit with it and it seems to work, I also tested it locally with `mvn clean install javadoc:javadoc -Predis` and all seems good :+1:  !

I will now integrated this in `master`.

Thank you very much for this contribution :pray:

**Comment by sbernard31** [comment]
(as I said we will ignore the github action issue, I haven't time to go deeper in it and didn't find any obvious way to fix it)

**Comment by aliakseiz** [comment]
> I think we're good now.

Thank you for all your advice and for patience :+1: 

I will start https://github.com/eclipse/leshan/issues/1249 now.

**Comment by sbernard31** [comment]
> Thank you for all your advice and for patience +1

You're welcome.

> I will start https://github.com/eclipse/leshan/issues/1249 now.

:+1:

**Comment by sbernard31** [comment]
This is now integrated in `master` (in squashed commit fd5c9c0b4c620e888fe8221c9142134c92292eed + typo commit b7dc3cf3f828864ff44d654891f15751f1a43668)

I notice that your email is `@any-ware.nl`. Fun fact, when I started to work on Eclipse project my company was named `anyware-technology`. It seems to be a common name in IoT :grin:

**Pull request closed without merging**